### PR TITLE
[T4A3][T09-B1]Bey Hao Yun

### PR DESCRIPTION
**Pull request by cardboardcode**
ready for review

### CODE REVIEW

File: src/seedu/addressbook/data/person/Contact.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package seedu.addressbook.data.person;
+
+/**
+ * Represents a Person's email,phone and address in the address book.
+ */
+
+public class Contact{
+	
+	public final String value;
```
Comment (cardboardcode) [inline src/seedu/addressbook/data/person/Contact.java]: Change to private attributes

File: src/seedu/addressbook/data/person/Email.java [hunk 2]
```diff
@@ -6,40 +6,31 @@
...
     /**
      * Validates given email.
      *
      * @throws IllegalValueException if given email address string is invalid.
      */
     public Email(String email, boolean isPrivate) throws IllegalValueException {
-        this.isPrivate = isPrivate;
+        super(email.trim(),isPrivate);
         email = email.trim();
         if (!isValidEmail(email)) {
```
Comment (cardboardcode) [inline src/seedu/addressbook/data/person/Email.java]: Redundant calling a private method which then calls the super class method "isValidEmail", instead just use isValidContact here.

File: src/seedu/addressbook/data/person/Phone.java [hunk 3]
```diff
@@ -6,39 +6,30 @@
...
      * Validates given phone number.
      *
      * @throws IllegalValueException if given phone string is invalid.
      */
     public Phone(String phone, boolean isPrivate) throws IllegalValueException {
-        this.isPrivate = isPrivate;
-        phone = phone.trim();
+    	super(phone.trim(), isPrivate);
+    	phone = phone.trim();
         if (!isValidPhone(phone)) {
```
Comment (cardboardcode) [inline src/seedu/addressbook/data/person/Phone.java]: Same point as the problem in Email.java

File: src/seedu/addressbook/data/person/Address.java [hunk 4]
```diff
@@ -6,38 +6,30 @@
...
 
     /**
      * Validates given address.
      *
      * @throws IllegalValueException if given address string is invalid.
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
-        this.isPrivate = isPrivate;
+        super(address.trim(), isPrivate);
         if (!isValidAddress(address)) {
```
Comment (cardboardcode) [inline src/seedu/addressbook/data/person/Address.java]: same problem as Email.java

File: src/seedu/addressbook/data/person/Address.java [hunk 5]
```diff
@@ -46,13 +38,4 @@ public boolean equals(Object other) {
                 || (other instanceof Address // instanceof handles nulls
                 && this.value.equals(((Address) other).value)); // state check
     }
-
-    @Override
-    public int hashCode() {
```
Comment (nirandiw) [inline src/seedu/addressbook/data/person/Address.java:51]: When you are overriding the equals() you must override the hashCode method too

**Comment by nirandiw** [comment]
Accepting this PR without a review because another of your PRs
for this tutorial was reviewed in detail.

**Pull request closed without merging**